### PR TITLE
fix query,use jacksonObjectMapper and view holder crash

### LINKED ISSUE #1637: Multiple Crashes And Fixes
**Describe the bug**
1.Incorrect Nested query #1345  
2.Use of ObjectMapper instead of jacksonObjectMapper() in #1592 
3.Crash in favorite event view holder because of null object reference in item_favourite

### PR DESCRIPTION
**Pull request by aggarwalpulkit596**
Fixes #1637 

Changes: Use jacksonObjectMapper instead of ObjectMapper(),Do not add chip view in favourtie view items,nested query fix for location events

Screenshots for the change:

### CODE REVIEW

**Comment by aggarwalpulkit596** [comment]
@iamareebjamal @nikit19 please review

**Comment by aggarwalpulkit596** [comment]
videotogif_2019 04 21_10 50 15

File: app/src/main/java/org/fossasia/openevent/general/event/EventViewHolder.kt [hunk 1]
```diff
@@ -61,17 +61,19 @@ class EventViewHolder(override val containerView: View) : RecyclerView.ViewHolde
...
-        }
-        event.eventTopic.let {
-            if (it != null)
-                addchips(it.name)
-        }
-        event.eventSubTopic.let {
-            if (it != null)
-                addchips(it.name)
+        if (containerView.chipTags != null) {
+            event.eventType.let {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/event/EventViewHolder.kt]:
```suggestion
            event.eventType?.let {
```

File: app/src/main/java/org/fossasia/openevent/general/event/EventViewHolder.kt [hunk 2]
```diff
@@ -61,17 +61,19 @@ class EventViewHolder(override val containerView: View) : RecyclerView.ViewHolde
...
+                if (it != null)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/event/EventViewHolder.kt]: Remove this

Comment (aggarwalpulkit596) [reply app/src/main/java/org/fossasia/openevent/general/event/EventViewHolder.kt]: If i remove this how i will be checking chipType  for favourite and similiar events cards

**Comment by aggarwalpulkit596** [comment]
@iamareebjamal updated

File: app/src/main/java/org/fossasia/openevent/general/event/EventViewHolder.kt [hunk 3]
```diff
@@ -61,17 +61,16 @@ class EventViewHolder(override val containerView: View) : RecyclerView.ViewHolde
...
-            if (it != null)
+        if (containerView.chipTags != null) {
+            event.eventType?.let {
                 addchips(it.name)
-        }
-        event.eventTopic.let {
-            if (it != null)
+            }
+            event.eventTopic?.let {
                 addchips(it.name)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/event/EventViewHolder.kt]: Wrong indentation

**Comment by aggarwalpulkit596** [comment]
@iamareebjamal fixing in a minute

File: app/src/main/java/org/fossasia/openevent/general/event/EventViewHolder.kt (hunk 3 above)
Comment (aggarwalpulkit596) [reply app/src/main/java/org/fossasia/openevent/general/event/EventViewHolder.kt]: @iamareebjamal Klint test would have failed if the indentation was wrong

**Comment by aggarwalpulkit596** [comment]
@iamareebjamal updated

**Pull request merged**